src/MangaJet/app/src/main/java/com/mangajet/mangajet/ui/favourite/FavouriteFragment.kt

```diff
@@ -1,15 +1,45 @@
 package com.mangajet.mangajet.ui.favourite
 
+import android.content.Intent
 import android.os.Bundle
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.fragment.app.setFragmentResult
+import com.google.android.material.bottomnavigation.BottomNavigationView
+import com.mangajet.mangajet.MangaJetApp
 import com.mangajet.mangajet.R
 import com.mangajet.mangajet.log.Logger
+import com.mangajet.mangajet.ui.history.HistoryFragment
 
 // Class which represents "Updates" fragment of MainActivity
 class FavouriteFragment : Fragment() {
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        when (requestCode) {
+            MangaJetApp.SEARCH_BY_TAG -> {
+                if (data == null) {
+                    super.onActivityResult(requestCode, resultCode, data)
+                    return
+                }
+                // send info for search
+
+                // collect info for search
+                val tag = data!!.getCharSequenceExtra("tag").toString()
+                val src = data!!.getCharSequenceExtra("src").toString()
+
+                MangaJetApp.isNeedToTagSearch = true
+                MangaJetApp.tagSearchInfo = Pair(tag, src)
```

Comment:
You can put this info into MangaJetApp in "AboutMangaFragment". This way you not needed to put data in Intent